### ISSUE TITLE
npm package doesn't have updated GraphiQLOptions for subscription config

### ISSUE THREAD

**Issue by ShaneKeney** [graphql/express-graphql#746]
Not sure where to report issue for npm package in itself so am reporting here.  The following is the type file on the latest npm install i.e.

```
npm install --save express-graphql
```

via. `renderGraphiQL.d.ts`
```
export interface GraphiQLOptions {
    /**
     * An optional GraphQL string to use when no query is provided and no stored
     * query exists from a previous session.  If undefined is provided, GraphiQL
     * will use its own default query.
     */
    defaultQuery?: string;
    /**
     * An optional boolean which enables the header editor when true.
     * Defaults to false.
     */
    headerEditorEnabled?: boolean;
}
```

Looking at the GitHub repo it seems the additional `subscriptionEndpoint?` field was added for subscription report.  Versioning on npm says `0.12.0` which matches with GitHub versioning but there seems to be a discrepancy and am getting some Typescript errors when trying to use subscriptions because of it.

**Comment by cowboyd** [comment]
@ShaneKeney It looks like the v0.12.0 release does not contain the improved subscription support. I've opened #746 to request the release of that awesome funcitonality.

**Comment by acao** [comment]
@ShaneKeney we merge to main and publish from main, so whatever is in main is not necessarily published. This is how graphql, react, and many oss projects work

**Comment by ShaneKeney** [comment]
@acao Is their an estimate as to when subscription support will be published?

**Comment by acao** [comment]
@junminstorage is working on the finishing touches once we’re able to publish graphiql next. We’re all volunteers so no timelines are known

**Comment by junminstorage** [comment]
@ShaneKeney , once the https://github.com/graphql/graphiql/pull/1806 is merged and changes are released, I will get PR https://github.com/graphql/express-graphql/pull/739 ready for review within a week or so.

**Comment by acao** [comment]
https://github.com/graphql/graphiql/runs/2314562284 it should finally publish when this is run @junminstorage, terribly sorry for the delays. I will be on later in the week!

**Comment by acao** [comment]
https://github.com/graphql/graphiql/blob/main/packages/graphiql/CHANGELOG.md#141 here's the release notes by the by! I may be able to create a PR for your PR myself later today, if that helps?

**Comment by junminstorage** [comment]
PR is here: https://github.com/graphql/express-graphql/pull/755. @acao can you take a look at it?

**Comment by junminstorage** [comment]
@ShaneKeney we release new version which should support your use case. let us know if it works for you.

**Comment by jeremyzahner** [comment]
@junminstorage Seems like no releases were made since November. Or am I missing something?

**Comment by ShaneKeney** [comment]
@junminstorage Same question as @jeremyzahner .  Doesn't seem to be any release since Novemeber.  Version 0.12.0 still does not have the support for subscription as explained above.  

I get the following error: 
```
Type '{ subscriptionEndpoint: string; }' is not assignable to type 'boolean | GraphiQLOptions | undefined'.
  Object literal may only specify known properties, and 'subscriptionEndpoint' does not exist in type 'GraphiQLOptions'.ts(2322
```

when the following is used: 

```
graphqlHTTP({
    schema,
    graphiql: { subscriptionEndpoint: `ws://localhost:${PORT}/subscriptions` },
})
```

**Comment by junminstorage** [comment]
@acao can you help with the release?

**Comment by rvalenciano** [comment]
Hello, 👋 Thank you very much for the volunteering time of the open-source contributors. I'm willing to contribute and donate time to this project if this means we can have the new release up; We also need it in our organization. Any plans on this or if my time can help speed things up for future releases?

**Comment by psugasti** [comment]
Would be great to have subscription support !

**Comment by acao** [comment]
In my free time, I'm quite busy with higher priority OSS work on graphiql at the moment, perhaps @danielrearden would be able to help?

I will hopefully get to this when I can, sorry for the delays

**Comment by acao** [comment]
my bad, i don't have permissions to publish this repo. @IvanGoncharov is the only active contributor that does

**Comment by acao** [comment]
To clarify, I’ve _never_ had publish permissions for this package. Only the ability to merge PRs

#764 will need to be completed first. I have put in a request for publish permissions again, hopefully they are granted this time!

**Comment by acao** [comment]
#764 is still blocked on some integration test suite issues I will address this weekend.

we also need to remove it from `graphiql/toolkit`'s optional dependencies, and cut another release of the `graphiql` monorepo.

**Comment by rvalenciano** [comment]
Hi @acao , any updates on this?

**Comment by acao** [comment]
sorry, there are higher priority issues in the graphql ecosystem right now, and we're all working for free! it will take some time

**Comment by rvalenciano** [comment]
Oh ok, well if any help is needed feel free to ping me. Best!

**Comment by junminstorage** [comment]
@acao @IvanGoncharov can you help with the release? If possible can you grant me or some of us here willing to help the permission to release?

**Comment by mhabegger** [comment]
Anyone still working on this? Is this project still maintained or dead?

**Comment by enisdenjo** [comment]
This library has been deprecated and this repo will be archived soon. It has been superseded by [`graphql-http`](https://github.com/graphql/graphql-http).

Furthermore, if you seek a fully-featured, well-maintained and performant server - I heavily recommend [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server)!
